src/Microsoft.SqlTools.ServiceLayer/Connection/ConnectionProviderOptionsHelper.cs

```diff
@@ -104,43 +105,76 @@ internal static ConnectionProviderOptions BuildConnectionProviderOptions()
+                        DisplayName = SR.ConnectionConfigOptions_applicationName_displayName,
+                        Description = SR.ConnectionConfigOptions_applicationName_description,
+                        ValueType = ConnectionOption.ValueTypeString,
+                        GroupName = GroupNames.General,
+                        SpecialValueType = ConnectionOption.SpecialValueAppName
                     },
                     new ConnectionOption
                     {
                         Name = "connectTimeout",
                         DisplayName = SR.ConnectionConfigOptions_connectTimeout_displayName,
                         Description = SR.ConnectionConfigOptions_connectTimeout_description,
                         ValueType = ConnectionOption.ValueTypeNumber,
-                        DefaultValue = "15",
+                        DefaultValue = "30",
                         GroupName = GroupNames.General
                     },
                     new ConnectionOption
                     {
-                        Name = "commandTimeout",
-                        DisplayName = SR.ConnectionConfigOptions_commandTimeout_displayName,
-                        Description = SR.ConnectionConfigOptions_commandTimeout_description,
-                        ValueType = ConnectionOption.ValueTypeNumber,
-                        DefaultValue = "30",
-                        GroupName = GroupNames.Initialization
+                        Name = "applicationIntent",
+                        DisplayName = SR.ConnectionConfigOptions_applicationIntent_displayName,
+                        Description = SR.ConnectionConfigOptions_applicationIntent_description,
+                        ValueType = ConnectionOption.ValueTypeCategory,
+                        CategoryValues =
+                        [
+                            new CategoryValue { Name = "ReadWrite", DisplayName = SR.ConnectionConfigOptions_applicationIntent_category_ReadWrite },
+                            new CategoryValue { Name = "ReadOnly", DisplayName = SR.ConnectionConfigOptions_applicationIntent_category_ReadOnly }
+                        ],
+                        GroupName = GroupNames.General
                     },
                     new ConnectionOption
                     {
-                        Name = "currentLanguage",
-                        DisplayName = SR.ConnectionConfigOptions_currentLanguage_displayName,
-                        Description = SR.ConnectionConfigOptions_currentLanguage_description,
-                        ValueType = ConnectionOption.ValueTypeString,
-                        GroupName = GroupNames.Initialization
+                        Name = "multiSubnetFailover",
+                        DisplayName = SR.ConnectionConfigOptions_multiSubnetFailover_displayName,
+                        ValueType = ConnectionOption.ValueTypeBoolean,
+                        GroupName = GroupNames.General
```

Comment:
I noticed in your screenshots that this is missing a description.  Wanna add one while you're here?

From the [SqlConnectionStringBuilder](https://learn.microsoft.com/en-us/dotnet/api/microsoft.data.sqlclient.sqlconnectionstringbuilder?view=sqlclient-dotnet-standard-5.2#properties) docs:
>If your application is connecting to an AlwaysOn availability group (AG) on different subnets, setting this to true provides faster detection of and connection to the (currently) active server.